src/simplicial-hott/09-yoneda.rzk.md

```diff
@@ -189,6 +190,105 @@ The Yoneda lemma says that evaluation at the identity defines an equivalence. Th
         ( evid-yon A is-segal-A a C is-covariant-C)))
 ```
 
+## Naturality
+
+The equivalence of the Yoneda lemma is natural in both $a : A$ and $C : A → U$.
+
+Naturality in $a$ follows from the fact that the maps `evid` and `yon` are
+fiberwise equivalences between covariant families over $A$, though it requires
+some work, which has not yet been formalized, to prove that the domain is
+covariant.
+
+Naturality in $C$ is not automatic but can be proven easily:
+
+```rzk title="RS17, Lemma 9.2(i)"
+#def is-natural-evid
+  (A : U)
+  (a : A)
+  (C D : A → U)
+  (ψ : (z : A) → C z → D z)
+  (φ : (z : A) -> hom A a z -> C z)
+  : (composition ((z : A) → hom A a z → C z) (C a) (D a) (ψ a) (evid A a C)) φ =
+    (composition ((z : A) → hom A a z → C z) ((z : A) → hom A a z → D z) (D a)
+    (evid A a D) ( \ α z g -> ψ z (α z g))) φ
+  := refl
+```
+
+```rzk title="RS17, Lemma 9.2(ii)"
+#def is-natural-yon-twice-pointwise
+  (A : U)
+  (is-segal-A : is-segal A)
+  (a : A)
+  (C D : A → U)
+  (is-covariant-C : is-covariant A C)
+  (is-covariant-D : is-covariant A D)
+  (ψ : (z : A) → C z → D z)
+  (u : C a)
+  (x : A)
+  (f : hom A a x)
+  : (composition (C a) (D a) ((z : A) → hom A a z → D z)
+      (yon A is-segal-A a D is-covariant-D) (ψ a)) u x f =
+    (composition (C a) ((z : A) → hom A a z → C z) ((z : A) → hom A a z → D z)
+      (\ α z g -> ψ z (α z g)) (yon A is-segal-A a C is-covariant-C)) u x f
+  := naturality-covariant-fiberwise-transformation
+      A a x f C D is-covariant-C is-covariant-D ψ u
+
+#def is-natural-yon-once-pointwise uses (funext)
+  (A : U)
+  (is-segal-A : is-segal A)
+  (a : A)
+  (C D : A → U)
+  (is-covariant-C : is-covariant A C)
+  (is-covariant-D : is-covariant A D)
+  (ψ : (z : A) → C z → D z)
+  (u : C a)
+  (x : A)
+  : (composition (C a) (D a) ((z : A) → hom A a z → D z)
+      (yon A is-segal-A a D is-covariant-D) (ψ a)) u x =
+    (composition (C a) ((z : A) → hom A a z → C z) ((z : A) → hom A a z → D z)
+      (\ α z g -> ψ z (α z g)) (yon A is-segal-A a C is-covariant-C)) u x
```

Comment:
```suggestion
      (\ α z g → ψ z (α z g)) (yon A is-segal-A a C is-covariant-C)) u x
```

Reply:
Whoops! I'd forgotten we'd switched to unicode arrows. Fixed now (I think).